apps/dsp-app/src/app/main/services/datadog-rum.service.ts

```diff
@@ -1,63 +1,67 @@
-                trackFrustrations: true,
-                useSecureSessionCookie: true,
-                beforeSend: (event, context) => {
-                    // collect a RUM resource's response headers
-                    if (
-                        event.type === 'resource' &&
-                        event.resource.type === 'xhr'
-                    ) {
-                        event.context = {
-                            ...event.context,
-                            responseHeaders: (
-                                context as RumFetchResourceEventDomainContext
-                            ).response?.body,
-                        };
-                    }
-                },
-            });
+    private buildTag$: Observable<BuildTag> = inject(BuildTagToken);
+    private config: DspInstrumentationConfig = inject(DspInstrumentationToken);
+    private session = inject(SessionService);
+    constructor() {
+        this.buildTag$.pipe(first()).subscribe((tag) => {
+            if (this.config.dataDog.enabled) {
+                datadogRum.init({
+                    applicationId: this.config.dataDog.applicationId,
+                    clientToken: this.config.dataDog.clientToken,
+                    site: this.config.dataDog.site,
+                    service: this.config.dataDog.service,
+                    env: this.config.environment,
+                    version: tag.build_tag,
+                    sessionSampleRate: 100,
+                    sessionReplaySampleRate: 100, // if not included, the default is 100
+                    trackResources: true,
+                    trackLongTasks: true,
+                    trackUserInteractions: true,
+                    trackFrustrations: true,
+                    useSecureSessionCookie: true,
+                    beforeSend: (event, context) => {
+                        // collect a RUM resource's response headers
+                        if (
+                            event.type === 'resource' &&
+                            event.resource.type === 'xhr'
+                        ) {
+                            event.context = {
+                                ...event.context,
+                                responseHeaders: (
+                                    context as RumFetchResourceEventDomainContext
+                                ).response?.body,
+                            };
+                        }
+                    },
+                });
 
-            // if session is valid: setActiveUser
-            this._s.isSessionValid().subscribe((response: boolean) => {
-                if (response) {
-                    const session: Session = this._s.getSession();
-                    this.setActiveUser(session.user.name, 'username');
-                }
-            });
-        }
+                // if session is valid: setActiveUser
+                this.session.isSessionValid().subscribe((response: boolean) => {
```

Comment:
Where do you unsubscribe?

Reply:
nowhere. This is already existing code and out of scope for this round.